custom_components/truenas/__init__.py

```diff
@@ -227,6 +234,33 @@ def device_info(self):
         }
 
 
+class TrueNASPoolEntity:
+    """Represents a pool on the TrueNAS host."""
+
+    _pool: CachingPool
```

Comment:
In order to support someone deleting a pool, this should be `Optional[CachingPool]`, unfortunately.  See how we do this for `TrueNASDiskEntity` above.